test/audienceVersionBump.test.ts

```diff
@@ -0,0 +1,120 @@
+import { sportKeywords } from './helpers/audienceDefinitions';
+import { edkt } from '../src';
+import {
+  clearStore,
+  getPageViews,
+  getMatchedAudiences,
+} from './helpers/localStorageSetup';
+import {
+  AudienceDefinition,
+  QueryFilterComparisonType,
+  StringArrayQueryValue,
+} from '../types';
+
+const sportPageFeature = {
+  keywords: {
+    version: 1,
+    value: sportKeywords,
+  },
+};
+
+type MakeAudienceData = {
+  version: number;
+  queryValue: StringArrayQueryValue;
+};
+
+export const makeAudience = ({
+  version,
+  queryValue,
+}: MakeAudienceData): AudienceDefinition => ({
+  id: 'iab-607',
+  version,
+  ttl: 3600,
+  lookBack: 0,
+  occurrences: 0,
+  definition: [
+    {
+      featureVersion: 1,
+      queryProperty: 'keywords',
+      queryValue,
+      queryFilterComparisonType: QueryFilterComparisonType.ARRAY_INTERSECTS,
+    },
+  ],
+});
+
+const runEdktWithData = (data: MakeAudienceData) =>
+  edkt.run({
+    pageFeatures: sportPageFeature,
+    audienceDefinitions: [makeAudience(data)],
+    omitGdprConsent: true,
+  });
+
+/* Should compute as follow:
+ * - if a user has matched version 1, and the most recent audience version available is 1, we should not run the checking.
+ * - if a user has matched version 1, and the most recent audience version available is 2, we should run the checking.
+ *   - if they not do not match the new version (2), but had matched 1, we must remove this audience from matched.
+ *   - if they matched v1 and now match on v2, they remain in the audience but with the updated version
+ */
+
+describe('edkt behaviour on audience version bump', () => {
+  describe('edkt unmatching behaviour on audience version bump', () => {
+    beforeAll(() => {
+      clearStore();
+    });
+
+    it('should match pageView against audienceDefinition', async () => {
+      await runEdktWithData({
+        version: 1,
+        queryValue: sportKeywords,
+      });
+
+      expect(getPageViews()).toHaveLength(1);
+      expect(getMatchedAudiences()).toHaveLength(1);
+    });
+
+    it('should skip checking on previously matched pageView/audienceDefinition', async () => {
+      await runEdktWithData({
+        version: 1,
+        queryValue: sportKeywords,
+      });
+
+      expect(getPageViews()).toHaveLength(2);
+      expect(getMatchedAudiences()).toHaveLength(1);
+    });
```

Comment:
I do not think this test achieves what it aims to achieve?

How do you know we skipped the engine check for that audience?

Reply:
It doesn't.
The skipping is an internal detail. The test just assures the no-op.
I got out of ideas for how to test the skipping without exposing unnecessary surface.

Do you have an idea for this or maybe is it better to remove this test?

Reply:
I think this test is more of a negative then benefit - it is named and gives confidence that something works when it does not test it.

One quick idea would be to mock (just on this run) an internal maybe the audience store - and spy on what gets called on it?

Reply:
The audience store always called, so the mock would not help.
I'll remove the test.
I like to use the test as a spec, but the skipping is an internal detail that does not affect the external behaviour.